File: AdventOfCode2023/Day 1/Day1.ts
import { readFileSync } from "fs";

const s = readFileSync("Day 1/input.txt", "utf-8").split("\n");
let total = 0;
for (let line = 0; line < s.length; line++) {
  let arr: string[] = [];
  for (let i = 0; i < s[line].length; i++) {
    if (+Number(s[line][i])) {
      arr.push(s[line][i]);
    }
  }

  total += 10 * Number(arr[0]) + Number(arr[arr.length - 1]);
}

console.log(total);

File: AdventOfCode2024/Day 1/Day1.py
fileContents = open("AdventOfCode2024/Day 1/input.txt")
arr = fileContents.read().split("\n")

left_numbers: list[int] = []
right_numbers: list[int] = []

for line in arr:
    left, right = int(line.split("   ")[0]), int(line.split("   ")[1])
    left_numbers.append(left)
    right_numbers.append(right)

left_numbers = sorted(left_numbers)
right_numbers = sorted(right_numbers)


total = 0
for i in range(len(left_numbers)):
    total += left_numbers[i] * len([j for j in right_numbers if j == left_numbers[i]])

print(total)